app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1372,15 +1373,12 @@ public function mergeLeads(Lead $lead, Lead $lead2, $autoMode = true)
         }
 
         //merge fields
-        $mergeFromFields = $mergeFrom->getFields();
-        foreach ($mergeFromFields as $group => $groupFields) {
-            foreach ($groupFields as $alias => $details) {
-                //overwrite old lead's data with new lead's if new lead's is not empty
-                if (!empty($details['value'])) {
-                    $mergeWith->addUpdatedField($alias, $details['value']);
-
-                    $this->logger->debug('LEAD: Updated '.$alias.' = '.$details['value']);
-                }
+        $mergeFromFields = $this->leadFieldModel->getFieldListWithProperties($mergeFrom);
+        foreach ($mergeFromFields as $mergeFromAlias => $mergeFromfield) {
+            //overwrite old lead's data with new lead's if new lead's is not empty
```

Comment:
Just at a quick read through, there's only one foreach here vs 2 in the original. Was this intended?

Reply:
It's ok.
Before fields were  loaded by   `$mergeFromFields = $mergeFrom->getFields();`
After PR  `$mergeFromFields = $this->leadFieldModel->getFieldListWithProperties($mergeFrom);`
Each function return different arrays, I replaced it, it's better readable.

Reply:
Okay, long as nothing was missed. Thanks!